src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -265,23 +279,46 @@ public void close() {
             }
         }
 
-        shutdownSharedExecutor(closeTimeout);
+        shutdownExecutor(sharedExecutor, closeTimeout, "rabbit-shared");
+        shutdownExecutor(channelChecker, closeTimeout, "channel-checker");
     }
 
     public void basicPublish(String exchange, String routingKey, BasicProperties props, byte[] body) throws IOException {
         ChannelHolder holder = getChannelFor(PinId.forRoutingKey(routingKey));
-        holder.withLock(channel -> {
-            channel.basicPublish(exchange, routingKey, props, body);
-        });
+        holder.withLock(channel -> channel.basicPublish(exchange, routingKey, props, body));
     }
 
     public SubscriberMonitor basicConsume(String queue, ManualAckDeliveryCallback deliverCallback, CancelCallback cancelCallback) throws IOException {
         ChannelHolder holder = getChannelFor(PinId.forQueue(queue));
         String tag = holder.mapWithLock(channel ->
                 channel.basicConsume(queue, false, subscriberName + "_" + nextSubscriberId.getAndIncrement(), (tagTmp, delivery) -> {
                     try {
-                        deliverCallback.handle(tagTmp, delivery,
-                                () -> holder.withLock(ch -> basicAck(ch, delivery.getEnvelope().getDeliveryTag())));
+                        Envelope envelope = delivery.getEnvelope();
+                        long deliveryTag = envelope.getDeliveryTag();
+                        String routingKey = envelope.getRoutingKey();
+                        LOGGER.trace("Received delivery {} from queue={} routing_key={}", deliveryTag, queue, routingKey);
+
+                        Confirmation confirmation = () -> holder.withLock(ch -> {
+                            try {
+                                basicAck(ch, deliveryTag);
+                            } finally {
+                                holder.release(() -> metrics.getReadinessMonitor().enable());
+                            }
+                        });
+
+                        holder.acquireAndSubmitCheck(() ->
```

Comment:
acquireAndSubmitCheck is called without lock, but it changes the holder's variable `check`